src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeParameterInfo.cs

```diff
@@ -126,21 +126,36 @@ public override
         {
             get
             {
-                if (ClassImpl == typeof(decimal) || ClassImpl == typeof(decimal?))
+                // In case of multiple CustomAttributes the following rules apply:
```

Comment:
I know you tried hard to match whatever nonsensical behavior exists in CoreCLR, but it is pretty hard.

If it helps, it would be fine to adjust the CoreCLR to make the ordering rules simpler and easier to replicate.